Form/Extension/BooleanExtension.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Form\Extension;
+
+use FOS\RestBundle\Form\Transformer\BooleanTypeToBooleanTransformer;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class BooleanExtension extends AbstractTypeExtension
+{
+    const TYPE_API = 'api';
+    const TYPE_CHECKBOX = 'checkbox';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['type'] === self::TYPE_API) {
```

Comment:
IMO the option name "type" is too generic. Can you find a better option name that describes better what the option does? E.g. `strict_string_handling`?